cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -351,8 +351,12 @@ private CloseableIterator<StructuredRow> getScanApplicationsIterator(StructuredT
     if (latestOnly) {
       // if only return latest version of the app, whether the range uses version field or creationTime field
       // doesn't matter, since records will be sort on appName
-      return table.scan(range, Integer.MAX_VALUE,
-                        Fields.stringField(StoreDefinition.AppMetadataStore.LATEST_FIELD, "true"), sortOrder);
+      // Also we treat latest=["true",null] as latest for pre 6.8 compatibility.
+      // So that UI can still show existing pipelines
```

Comment:
nit: it's not really because of the UI, and CDAP only knows about 'applications' and not 'pipelines'. Let's change this line to:

Prior to 6.8, all versions of an application were returned in the list of applications, not just the latest version.

Reply:
got it, updated the comment. Thanks!